docs/versioned_docs/version-zenith/guides/457482-create-app-ci-module.mdx

```diff
@@ -0,0 +1,350 @@
+
+## Step 1: Initialize a new module
+
+The example module used in this guide builds, tests and publishes a Node.js application.
+
+Within your application directory, create a new `ci/` directory:
+
+```shell
+mkdir ci && cd ci
+```
+
+Run `dagger mod init` in the `ci` directory to bootstrap a new module:
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+```sh
+dagger mod init --name=my-module --sdk=go
+```
+
+This will generate a `dagger.json` module file, an initial `main.go` source file, as well as a generated `dagger.gen.go` and `querybuilder` directory for the generated module code.
+
+</TabItem>
+<TabItem value="Python">
+
+```sh
+dagger mod init --name=my-module --sdk=python
+```
+
+This will generate a `dagger.json` module file, initial `src/main.py` and `pyproject.toml` files, as well as a generated `sdk` folder for local development.
+
+</TabItem>
+<TabItem value="TypeScript">
+
+```sh
+dagger mod init --name=my-module --sdk=typescript
+```
+
+This will generate a `dagger.json` module file, initial `src/index.ts`, `package.json` and `tsconfig.json` files, as well as a generated `sdk` folder for local development.
+
+</TabItem>
+</Tabs>
+
+## Step 2: Add a function to build the application base image
+
+The first step is to add a function to build a base image containing the application source code and runtime. This base image will serve as an input to other functions.
+
+Since the application is a Node.js application, it's convenient to use the [`node` module](https://github.com/quartz-technology/daggerverse/node), which provides a set of ready-made functions to manage a Node.js project.
+
+:::note
+Dagger exposes every module using a language-agnostic HTTP API. So, even though the [`node` module](https://github.com/quartz-technology/daggerverse/node) is written in TypeScript, you can transparently call its functions from your module written in Go, Python or any other supported language.
+:::
+
+First, add the `node` module as a dependency:
+
+```shell
+dagger mod install github.com/quartz-technology/daggerverse/node/typescript
+```
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+Next, update the generated `main.go` file with the following code:
+
+```go file=./snippets/create-app-ci-module/base/go/main.go
+```
+
+</TabItem>
+<TabItem value="Python">
+
+Next, update the generated `src/main.py` file with the following code:
+
+```python file=./snippets/create-app-ci-module/base/python/main.py
+```
+
+</TabItem>
+<TabItem value="TypeScript">
+
+Next, update the generated `src/index.ts` file with the following code:
+
+```typescript file=./snippets/create-app-ci-module/base/typescript/index.ts
+```
+
+</TabItem>
+</Tabs>
+
+This function does the following:
+
+  - It calls the `node` module's `WithVersion()` API via the `dag` client. This API method returns a `node` container image with the given Node.js version. This container image is represented as a `Node` object.
```

Comment:
Instead of having 3 versions of this, one for each tab with minor differences on account of the function names, I'm trying to have a single explanation and using the API method instead. Any feedback on this approach (good or revert to previous) would be appreciated.

Also, checking terminology, which is best: `XX API`, `XX API method`, `XX function`...? Something else?

Reply:
In GraphQL these are called "fields". Also, it follows Javascript naming conventions: "PascalCase" for types and "camelCase" for fields and arguments.

We could create a component with inputs being the API name and (activated) SDK, which would produce the converted name. Or something like this that can author with a single source of truth but produce the replaced values at build time.

Reply:
I'm not sure if there are available cycles in the team atm to create such a component. In its absence, it sounds like you find the current approach less readable than the previous one, which had the same text repeated for each tab and customized to each language SDK?